src/include/hip/rpp_hip_common.hpp

```diff
@@ -1826,6 +1826,18 @@ __device__ __forceinline__ void rpp_hip_math_bitwiseOr8(d_float8 *src1_f8, d_flo
         dst_f8->f1[7] = (float)((uchar)(src1_f8->f1[7]) | (uchar)(src2_f8->f1[7]));
 }
 
+__device__ __forceinline__ void rpp_hip_math_exclusiveOr8(d_float8 *src1_f8, d_float8 *src2_f8, d_float8 *dst_f8)
+{
+        dst_f8->f1[0] = (float)((uchar)(std::nearbyintf)(src1_f8->f1[0]) ^ (uchar)(std::nearbyintf)(src2_f8->f1[0]));
```

Comment:
I don't think you should round the values when doing exclusive or. Please use a reinterpret_cast<int> and do a simple XOR between int values

Reply:
- The logical operations (XOR, Bitwise And etc) support is provided for images only. Images can be stored in the following formats and carry the following values for the formats : U8 (0 to 255), I8 (-128 to 127), Float (0 to 1)
- In case of float image, we rescale the images to 0 to 255 range before we perform the logical operations. While rescaling, some of the values are close to the ceil value (Eg - 224.99267578125 in case of representing 225) and to account for the same rounding operation is done here. Usage of reinterpret cast and performing the operation leads to output mismatches in this case. Thanks